common/chunkStatusLogger.go

```diff
@@ -315,6 +318,13 @@ func (csl *chunkStatusLogger) FlushLog() {
 	}
 }
 
+// CloseLogger close the chunklogger thread.
+func (csl *chunkStatusLogger) CloseLogger() {
+	csl.outputEnabled = false
+	csl.done <- struct{}{}
+	close(csl.unsavedEntries)
```

Comment:
Can we please help the reader by writing some useful comments.
f.e.

        // Once logger is closed, we log no more chunks. 
	csl.outputEnabled = false

        // Signal main logger goroutine to packup.
	csl.done <- struct{}{}

        // No more chunks will ever be written, let the main logger know about this.
	close(csl.unsavedEntries)